infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -292,7 +287,9 @@ public synchronized void handleMessageLayoutPropose(CorfuPayloadMsg<LayoutPropos
 
         long serverEpoch = getServerEpoch();
 
-        if (msg.getPayload().getEpoch() != serverEpoch) {
+        // If both epoch numbers in the msg aren't equal, then the request is malformed.
+        if (msg.getPayload().getEpoch() != msg.getPayload().getLayout().getEpoch() ||
```

Comment:
I suggest to patch this PR with the necessary additions/fixes, before merging.

On the runtime side, the closest I can find is UnprocessedException, which is actually seems to be over/ab-used in many places to indicate "unsupported". Perhaps this needs cleaning up as well.

On the server side, I see a need for a generic "malformed/rejected" error reply by any server. This happens when a runtime somehow generates a bogus message. I think we may want it not to be specific to any component (layout, logunit, etc.), but a wildcard rejection that instructs the client to drop.